pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/name/SimpleSegmentNameGenerator.java

```diff
@@ -50,6 +50,12 @@ public SimpleSegmentNameGenerator(String segmentNamePrefix, @Nullable String seg
 
   @Override
   public String generateSegmentName(int sequenceId, @Nullable Object minTimeValue, @Nullable Object maxTimeValue) {
+    if (minTimeValue != null) {
+      minTimeValue = minTimeValue.toString().replaceAll("[: \\/]", "_");
```

Comment:
It's better to precompile these i.e. 

```java
private static final Pattern REPLACEMENT_REGEX = Pattern.compile("[: \\/]");

...

minTimeValue = minTimeValue.toString().replaceAll(REPLACEMENT_REGEX, "_");
```

This isn't a very hot code path but always doing this will make it easier to ban on the fly regex compilation with static analysis.

Reply:
got it, have updated it.